app/controllers/api/subcollections/vms.rb

```diff
@@ -27,6 +20,21 @@ def vms_query_resource(object)
           vm.as_json.merge(conflictless_hash)
         end
       end
+
+      private
+
+      def create_vm_attributes_hash(vm_attrs, vm)
+        vm_attrs.each_with_object({}) do |attr, hash|
+          hash[attr] = vm.public_send(attr.to_sym) if vm.respond_to?(attr.to_sym)
+        end.compact
+      end
+
+      def create_vm_decorators_hash(vm_decorators, vm)
+        decorated = vm.decorator_class? ? vm.decorate : nil
+        vm_decorators.each_with_object({}) do |name, hash|
+          hash[name] = vm.decorate.public_send(name.to_sym) if decorated.respond_to?(name.to_sym)
+        end.compact unless decorated.nil?
```

Comment:
Having a postfix conditional at the end of a multiline block can be disruptive to follow. WDYT about turning this into a guard clause instead?